src/transform-img/transform-img.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Curve} from '../bezier-curve-utils.js';
+import {getRenderedDimensions} from '../img-dimensions.js';
+import {createReplacement} from '../replacement-img.js';
+import {prepareCropAnimation} from './crop-animation.js';
+import {prepareScaleAnimation} from './scale-animation.js';
+import {prepareTranslateAnimation} from './translate-animation.js';
+
+/**
+ * A counter that makes sure the keyframes names are unique.
+ */
+let keyframesPrefixCounter: number = 0;
+
+/**
+ * Gets a prefix name to use for the keyframes, avoiding clashing with other
+ * keyframes that may be defined.
+ * @param namespace A namespace to use as a prefix.
+ * @return The prefix to use for the various keyframes.
+ */
+function getKeyframesPrefix(namespace: string): string {
+  keyframesPrefixCounter += 1;
+
+  return `${namespace}-${keyframesPrefixCounter}-`;
+}
+
+/**
+ * Prepares an animation from one image to another. Creates a a temporary
+ * replacement image that is transitioned between the two images.
+ * @param options
+ * @param options.transitionContainer The container to place the transition
+ *    image in. Defaults to document.body.
+ * @param options.styleContainer The container to place the generated
+ *    stylesheet in. Defaults to document.head.
+ * @param options.srcImg The image to transition from.
+ * @param options.targetImg The image to transition to.
+ * @param options.srcImgRect The ClientRect for where the transition should
+ *    start from. Defaults to the current rect for srcImg.
+ * @param options.targetImgRect The ClientRect for where the transition should
+ *    end at. Defaults to the current rect for targetImg.
+ * @param options.curve Control points for a Bezier curve to use for the
```

Comment:
Per discussion, changed to use the same styles and curve for all the animations since that seems to be the only thing that will make everything line up in the animation correctly.

Added a default curve of `ease-in-out`.